### PR TITLE
fix: add default constructor for mongo

### PR DESCRIPTION
**Pull request by pietro-tota**
#### List of Changes

Add persistence constructor used by Mongo data for events serialization/deserialization

#### Motivation and Context

This modification is needed in order to make spring-data working correctly

#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/pagopa/ecommerce/commons/domain/AESMetadata.java [hunk 1]
```diff
@@ -45,6 +46,11 @@ private AESMetadata(
         this(Base64.getDecoder().decode(salt), Base64.getDecoder().decode(iv)); // NOSONAR
     }
 
+    @PersistenceConstructor
+    public AESMetadata {
+        //used by Mongo for serialization
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/ecommerce/commons/domain/AESMetadata.java:51]:
```suggestion
        // used by Mongo for deserialization
```

**Review by giovanniberti — DISMISSED** [review]

File: src/main/java/it/pagopa/ecommerce/commons/domain/AESMetadata.java (hunk 1 above)
Comment (pietro-tota) [reply src/main/java/it/pagopa/ecommerce/commons/domain/AESMetadata.java]: fixed

**Comment by giovanniberti** [comment]
You may need to add `@SuppressWarnings` for Sonar's code smell about empty constructors, as [here](https://github.com/pagopa/pagopa-ecommerce-commons/blob/d894635b40ec03d09bdf512edd4b9ff8ae3a94b5/src/main/java/it/pagopa/ecommerce/commons/domain/Confidential.java#L36-L53) :)

**Comment by pietro-tota** [comment]
> You may need to add `@SuppressWarnings` if Sonar complains about empty constructors, as [here](https://github.com/pagopa/pagopa-ecommerce-commons/blob/d894635b40ec03d09bdf512edd4b9ff8ae3a94b5/src/main/java/it/pagopa/ecommerce/commons/domain/Confidential.java#L36-L53) :)

Yes, missed it in previous commit. Done😄

**Review by giovanniberti — APPROVED** [review]

**Pull request merged**